liquibase-core/src/main/java/liquibase/util/StringUtils.java

```diff
@@ -88,12 +88,80 @@ public static String[] processMultiLineSQL(String multiLineSQL, boolean stripCom
 
         String trimmedString = StringUtils.trimToNull(currentString.toString());
         if (trimmedString != null) {
-            returnArray.add(trimmedString);
+            if (!returnArray.isEmpty() && isLastStatementComment(parsed, trimmedString)) {
```

Comment:
I like that you have the new tests to check that an input string with a comment in the end is correctly parsed. From that, I can tell that this "if" block at the end of this function does strip it off. But it does feel like a bit of an odd spot for it.

In the code above this you are already iterating through the tokens and building up the currentString. It seems better to have logic in there to not append comments in the first place (if the user asked for them not to be included) than to have to do a bit of reparsing/rebuilding of the currentString at this point? 

OR: we maybe should even be making the change in this code in the first place? Line 41 of `StringClauses parsed = SqlParser.parse(multiLineSQL, true, !stripComments);` is supposed to be only preserving comments if stripComments=false, so in theory you shouldn't have comments to do deal with in this function at all. If the problem is that `SqlParser.parse` is not correctly stripping comments when the comment is at the end of the string then it should be that function that gets fixed up.

Reply:
SqlParser works correctly. If user asks not to append comments, SqlParser will strip them out and logic above won't see any comments. So new logic which I added in this PR will change behaviour only if customer does want to keep comments. 

Bug itself is part of the function which splits StringClauses into statements, because if comment is in the middle of SQL script it is part of the statement and JDBC is good with it, but if comment is last and it is split into separate statement JDBC throws an error when executes it because it does not make any changes on DB. That's why I added logic which checks if the last statement is a pure SQL comment and if it - then we just append it to the previous statement.

Reply:
That make sense, thanks for clarifying the bug